### PR TITLE
fix: dup. column mapping property check on update

### PR DESCRIPTION
**Pull request by martinmaillard**
Fix duplicate target property column mapping check on update

### CODE REVIEW

File: apis/core/lib/domain/column-mapping/update.ts [hunk 1]
```diff
@@ -50,10 +50,12 @@ export async function updateColumnMapping({
...
-      throw new DomainError(`Target property already mapped from column ${column?.name}`)
+    if (table.isObservationTable) {
+      const mappingExists = await findMapping(table, targetProperty, store)
+      if (mappingExists) {
+        const column = await store.getResource<CsvColumn>(mappingExists.sourceColumn?.id)
+        throw new DomainError(`Target property already mapped from column ${column?.name}`)
+      }
     }
 
     if (table.types.has(cc.ObservationTable)) {
```
Comment (lucafurrer) [inline apis/core/lib/domain/column-mapping/update.ts:61]: If we check for the observation table, I would merge both checks for the observation tables , no?
```suggestion
    if (table.isObservationTable) {
      const mappingExists = await findMapping(table, targetProperty, store)
      if (mappingExists) {
        const column = await store.getResource<CsvColumn>(mappingExists.sourceColumn?.id)
        throw new DomainError(`Target property already mapped from column ${column?.name}`)
      }
```

**Review by lucafurrer — COMMENTED** [review]
We do not need unicity for the targetProperty outside observation tables?

**Comment by martinmaillard** [comment]
Judging by the discussions, I'm not sure we need unicity at all. But here I'm just replicating what was implemented in "createColumnMapping".

File: apis/core/lib/domain/column-mapping/update.ts (hunk 1 above)
Comment (martinmaillard) [reply apis/core/lib/domain/column-mapping/update.ts:61]: Be my guest ;) I don't have much time to work on this. I just noticed that the behavior was not consistent between "create" and "update" and quickly wanted to fix it

Comment (lucafurrer) [reply apis/core/lib/domain/column-mapping/update.ts]: Here you go :wink:

**Pull request merged**